bundles/org.eclipse.swt/Eclipse SWT Drag and Drop/gtk/org/eclipse/swt/dnd/TextTransfer.java

```diff
@@ -111,26 +111,26 @@ public void javaToNative (Object object, TransferData transferData) {
 		transferData.pValue = string_target;
 		transferData.result = 1;
 	}
-    if (transferData.type == TEXT_PLAIN_UTF8_ID) {
-        // Convert the text into RFC-1341 format
-        byte[] rfc1341Data = encodeTextAsRFC1341(string);
-        transferData.format = 8; // Format for UTF-8
-        transferData.length = rfc1341Data.length;
-        transferData.pValue = OS.g_malloc(rfc1341Data.length);
-        if (transferData.pValue != 0) {
-            C.memmove(transferData.pValue, rfc1341Data, rfc1341Data.length);
-            transferData.result = 1;
-        }
-    }
+	if (transferData.type == TEXT_PLAIN_UTF8_ID) {
+		// Convert the text into RFC-1341 format
+		byte[] rfc1341Data = encodeTextAsRFC1341(string);
+		transferData.format = 8; // Format for UTF-8
+		transferData.length = rfc1341Data.length;
+		transferData.pValue = OS.g_malloc(rfc1341Data.length);
+		if (transferData.pValue != 0) {
+			C.memmove(transferData.pValue, rfc1341Data, rfc1341Data.length);
+			transferData.result = 1;
+		}
+	}
 }
 
 // New method to encode text as RFC-1341
 private byte[] encodeTextAsRFC1341(String text) {
-    // Implement encoding logic here, e.g., adding MIME headers and encoding text
-    // This is a simplified example; actual encoding depends on RFC-1341 standards
-//    String rfc1341Text = "Content-Type: " + TEXTPLAINUTF8 + "\r\n\r\n" + text;
-    String rfc1341Text = text;
-    return rfc1341Text.getBytes(StandardCharsets.UTF_8);
+	// Implement encoding logic here, e.g., adding MIME headers and encoding text
+	// This is a simplified example; actual encoding depends on RFC-1341 standards
+	//    String rfc1341Text = "Content-Type: " + TEXTPLAINUTF8 + "\r\n\r\n" + text;
```

Comment:
If you are cleaning code, can we please remove this commented line 
`// String rfc1341Text = "Content-Type: " + TEXTPLAINUTF8 + "\r\n\r\n" + text;`
and change the next two lines to simply return `text.getBytes(StandardCharsets.UTF_8);`.


But looking at this method (I haven't reviewed original PR) and at what was implemented before, I wonder why don't you use 
`byte[] utf8 = Converter.wcsToMbcs (string, true);` as it was done before and avoid converting String to String which seem to be wrong to me.

Reply:
> If you are cleaning code, can we please remove this commented line
> // String rfc1341Text = "Content-Type: " + TEXTPLAINUTF8 + "\r\n\r\n" + text;
> and change the next two lines to simply return text.getBytes(StandardCharsets.UTF_8);.
OK the commented line led to the string "text/plain;charset=utf-8" being prepended when pasting from Eclipse into for example kwrite.

> I wonder why don't you use
> byte[] utf8 = Converter.wcsToMbcs (string, true); as it was done before and avoid converting String to String which seem to be wrong to me.0

It is supposed to convert  uft-8 to string.see https://github.com/eclipse-platform/eclipse.platform.swt/pull/863#discussion_r1389349810

Ok seems like a sensible suggestion

[Re: [platform-swt-dev] Correct way to translate Java Strings ...](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwjS1PuSvbmCAxWOQfEDHRA9Ds0QFnoECB0QAQ&url=https%3A%2F%2Fwww.eclipse.org%2Flists%2Fplatform-swt-dev%2Fmsg08079.html&usg=AOvVaw1eofBRP9kD6R6eGnX_DL9-&opi=89978449)

However this then changes the PR status to functional change from non functional.
The crucial quote here is:

> It seems that using getbytes(..) can lead to issues if one passes the result to a call that expects a null-terminated String.

Reply:
I will have to do some functional testing before pushing the functional change suggestion.

Reply:
@iloveeclipse 
BTW I don't know how to test for occurrences of null terminated strings.
But I can test if there are issues with the danish special characters æøå